ratis-server-api/src/main/java/org/apache/ratis/server/raftlog/RaftLogSequentialOps.java

```diff
@@ -128,9 +128,11 @@ <OUTPUT, THROWABLE extends Throwable> OUTPUT runSequentially(
    * Used by the leader.
    */
   default CompletableFuture<Long> appendEntry(LogEntryProto entry, TransactionContext context) {
-    return appendEntry(entry);
+    return appendEntry(ReferenceCountedObject.wrap(entry), context);
   }
 
+  CompletableFuture<Long> appendEntry(ReferenceCountedObject<LogEntryProto> entryRef, TransactionContext context);
+
```

Comment:
Since this is in `ratis-server-api`, we have to provide a default implementation of the new method for compatibility.
```java
  /**
   * Append asynchronously an entry.
   * Used by the leader.
   */
  default CompletableFuture<Long> appendEntry(ReferenceCountedObject<LogEntryProto> entryRef, TransactionContext context) {
    return appendEntry(entryRef.get(), context);
  }

  /**
   * @deprecated use {@link #append(ReferenceCountedObject)}.
   */
  @Deprecated
  default CompletableFuture<Long> appendEntry(LogEntryProto entry, TransactionContext context) {
    throw new UnsupportedOperationException();
  }
```

Reply:
Update also `RaftLogBase`:
```java
@@ -345,15 +346,17 @@ public abstract class RaftLogBase implements RaftLog {
 
   @Override
   public final CompletableFuture<Long> appendEntry(LogEntryProto entry) {
-    return appendEntry(entry, null);
+    return appendEntry(ReferenceCountedObject.wrap(entry), null);
   }
 
```